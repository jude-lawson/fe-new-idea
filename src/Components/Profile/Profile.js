import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export class Profile extends Component {

  render() {
    const { user } = this.props;
    return (
      <section>
        {console.log(user)}
      </section>
    );
  }
}

Profile.propTypes = {
  user: PropTypes.object
};

const mapStateToProps = state => ({
  user: state.user
});

const mapDispatchToProps = () => ({

});

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
