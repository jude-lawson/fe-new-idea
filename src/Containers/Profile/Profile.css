.profile-card {
  align-items: center;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 35rem;
  position: relative;
}

.profile-card__user {
  background: var(--primary-white);
  color: var(--primary-dark-grey);
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  bottom: -15rem;
  position: absolute;
  text-align: center;
  width: 100%;
  height: 15rem;
  padding: 2rem;
}
