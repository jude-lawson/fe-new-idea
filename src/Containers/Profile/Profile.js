import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Header from '../Header/Header';

import './Profile.css';

export class Profile extends Component {

  componentDidMount() {
    
  }

  render() {
    const { user } = this.props;
    const userCardStyles = {
      background: `no-repeat url(${user.image}) center center`,
      height: '60vh'
    };

    return (
      <div>
        <Header />
        <section className="profile-container">
          {user.id &&
          <div className="profile-card" style={userCardStyles}>
            <div className="profile-card__user">
              <h2>{user.name}</h2>
            </div>
          </div>
          }
        </section>
      </div>
    );
  }
}

Profile.propTypes = {
  user: PropTypes.object
};

export const mapStateToProps = state => ({
  user: state.user,
  ideas: state.ideas
});

export default connect(mapStateToProps)(Profile);
