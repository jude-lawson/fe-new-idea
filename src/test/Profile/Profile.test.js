import React from 'react';
import { shallow } from 'enzyme';
import { Profile, mapStateToProps, mapDispatchToProps } from '../../Containers/Profile/Profile';

describe('<Profile />', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<Profile />));

  test('should user to state', () => {

  });

  test('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  });
});
