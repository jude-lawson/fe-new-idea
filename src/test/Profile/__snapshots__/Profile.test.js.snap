// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should not render a user when none present 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <section
          className="profile-container"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "profile-container",
        },
        "ref": null,
        "rendered": null,
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <section
            className="profile-container"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "profile-container",
          },
          "ref": null,
          "rendered": null,
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;

exports[`<Profile /> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    user={
        Object {
            "id": "3214adf",
            "image": "some-url-here",
            "name": "Joben",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <section
          className="profile-container"
>
          <div
                    className="profile-card"
                    style={
                              Object {
                                        "background": "no-repeat url(some-url-here) center center",
                                        "height": "50vh",
                                      }
                    }
          >
                    <div
                              className="profile-card__user"
                    >
                              <h2>
                                        Joben
                              </h2>
                    </div>
          </div>
</section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="profile-card"
            style={
                        Object {
                                    "background": "no-repeat url(some-url-here) center center",
                                    "height": "50vh",
                                  }
            }
>
            <div
                        className="profile-card__user"
            >
                        <h2>
                                    Joben
                        </h2>
            </div>
</div>,
          "className": "profile-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="profile-card__user"
>
              <h2>
                            Joben
              </h2>
</div>,
            "className": "profile-card",
            "style": Object {
              "background": "no-repeat url(some-url-here) center center",
              "height": "50vh",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2>
                Joben
</h2>,
              "className": "profile-card__user",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Joben",
              },
              "ref": null,
              "rendered": "Joben",
              "type": "h2",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <section
            className="profile-container"
>
            <div
                        className="profile-card"
                        style={
                                    Object {
                                                "background": "no-repeat url(some-url-here) center center",
                                                "height": "50vh",
                                              }
                        }
            >
                        <div
                                    className="profile-card__user"
                        >
                                    <h2>
                                                Joben
                                    </h2>
                        </div>
            </div>
</section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="profile-card"
              style={
                            Object {
                                          "background": "no-repeat url(some-url-here) center center",
                                          "height": "50vh",
                                        }
              }
>
              <div
                            className="profile-card__user"
              >
                            <h2>
                                          Joben
                            </h2>
              </div>
</div>,
            "className": "profile-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="profile-card__user"
>
                <h2>
                                Joben
                </h2>
</div>,
              "className": "profile-card",
              "style": Object {
                "background": "no-repeat url(some-url-here) center center",
                "height": "50vh",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h2>
                  Joben
</h2>,
                "className": "profile-card__user",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Joben",
                },
                "ref": null,
                "rendered": "Joben",
                "type": "h2",
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
